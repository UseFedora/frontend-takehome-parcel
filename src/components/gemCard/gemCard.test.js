import React from 'react';
import { shallow } from 'enzyme';
import Constants from '../../common/constants';
import GemCard from './';

describe('GemCard', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <GemCard
        name={"Railties"}
        version={"1.0.1"}
        info={"Does stuff"}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should output the supplied information', () => {
    const wrapper = shallow(
      <GemCard
        name={"Railties"}
        version={"1.0.1"}
        info={"Does stuff"}
      />
    );
    expect(wrapper.text()).toContain('Railties');
    expect(wrapper.text()).toContain('1.0.1');
    expect(wrapper.text()).toContain('Does stuff');
  });

  it('should not output empty fields if no values supplied', () => {
    const wrapper = shallow(
      <GemCard
        name={"Railties"}
        version={"1.0.1"}
        info={"Does stuff"}
      />
    );
    expect(wrapper.text()).not.toContain('Authors(s):');
    expect(wrapper.text()).not.toContain('Licenses(s):');
    expect(wrapper.text()).not.toContain('Project:');
    expect(wrapper.text()).not.toContain('Gem:');
    expect(wrapper.text()).not.toContain('Homepage:');
    expect(wrapper.text()).not.toContain('Documentation:');
  });

  it('should output the author if it\'s been supplied', () => {
    const wrapper = shallow(
      <GemCard
        name={"Railties"}
        version={"1.0.1"}
        info={"Does stuff"}
        authors={"William Shakespeare"}
      />
    );
    expect(wrapper.text()).toContain('Authors(s):');
    expect(wrapper.text()).toContain('William Shakespeare');
  });

  it('should output the licenses if they have been supplied', () => {
    const wrapper = shallow(
      <GemCard
        name={"Railties"}
        version={"1.0.1"}
        info={"Does stuff"}
        licenses={['MIT', 'Apache']}
      />
    );
    expect(wrapper.text()).toContain('License(s):');
    expect(wrapper.text()).toContain('MIT, Apache');
  });

  it('should output the project URI if it\'s been supplied', () => {
    const wrapper = shallow(
      <GemCard
        name={"Railties"}
        version={"1.0.1"}
        info={"Does stuff"}
        project_uri={"www.rubygems.org/project"}
      />
    );
    expect(wrapper.text()).toContain('Project:');
    expect(wrapper.text()).toContain('www.rubygems.org/project');
  });

  it('should output the gem URI if it\'s been supplied', () => {
    const wrapper = shallow(
      <GemCard
        name={"Railties"}
        version={"1.0.1"}
        info={"Does stuff"}
        gem_uri={"www.rubygems.org/gem"}
      />
    );
    expect(wrapper.text()).toContain('Gem:');
    expect(wrapper.text()).toContain('www.rubygems.org/gem');
  });

  it('should output the homepage URI if it\'s been supplied', () => {
    const wrapper = shallow(
      <GemCard
        name={"Railties"}
        version={"1.0.1"}
        info={"Does stuff"}
        homepage_uri={"www.rubygems.org"}
      />
    );
    expect(wrapper.text()).toContain('Homepage:');
    expect(wrapper.text()).toContain('www.rubygems.org');
  });

  it('should output the documentation URI if it\'s been supplied', () => {
    const wrapper = shallow(
      <GemCard
        name={"Railties"}
        version={"1.0.1"}
        info={"Does stuff"}
        documentation_uri={"www.rubygems.org/docs"}
      />
    );
    expect(wrapper.text()).toContain('Documentation:');
    expect(wrapper.text()).toContain('www.rubygems.org/docs');
  });

  it('should register a click on the save/unsave icon', () => {
    const spyOnClickSave = jest.fn();

    const wrapper = shallow(
      <GemCard
        name={"Railties"}
        version={"1.0.1"}
        info={"Does stuff"}
        onClickSave={spyOnClickSave}
      />
    );
    wrapper.find('.save-icon-test-target').simulate('click');
    expect(spyOnClickSave).toBeCalled();
  });

  it('should call the handler with the save action', () => {
    const spyOnClickSave = jest.fn();

    const wrapper = shallow(
      <GemCard
        name={"Railties"}
        version={"1.0.1"}
        info={"Does stuff"}
        onClickSave={spyOnClickSave}
      />
    );
    wrapper.find('.save-icon-test-target').simulate('click');
    expect(spyOnClickSave).toBeCalledWith(Constants.UiElements.ACTION_SAVE, 'Railties');
  });

  it('should call the handler with the unsave action', () => {
    const spyOnClickSave = jest.fn();

    const wrapper = shallow(
      <GemCard
        name={"Railties"}
        version={"1.0.1"}
        info={"Does stuff"}
        isSaved={true}
        onClickSave={spyOnClickSave}
      />
    );
    wrapper.find('.save-icon-test-target').simulate('click');
    expect(spyOnClickSave).toBeCalledWith(Constants.UiElements.ACTION_DELETE_SAVED, 'Railties');
  });
});
