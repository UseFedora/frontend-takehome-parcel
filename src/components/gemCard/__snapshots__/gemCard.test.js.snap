// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GemCard should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GemCard
    authors=""
    documentation_uri=""
    downloads={0}
    gem_uri=""
    homepage_uri=""
    info="Does stuff"
    isSaved={false}
    licenses={Array []}
    name="Railties"
    onClickSave={[Function]}
    project_uri=""
    version="1.0.1"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className=", saveIconContainer"
          onClick={[Function]}
        >
          <FaRegGem
            className=" "
          />
        </div>,
        <div>
          <div>
            <div>
              <span>
                Railties
              </span>
              <span>
                  (v. 
                1.0.1
                )
              </span>
            </div>
            <div>
              0
               
              <span>
                downloads
              </span>
            </div>
          </div>
          <div>
            Does stuff
          </div>
        </div>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <FaRegGem
            className=" "
          />,
          "className": ", saveIconContainer",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": " ",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <div>
                <span>
                  Railties
                </span>
                <span>
                    (v. 
                  1.0.1
                  )
                </span>
              </div>
              <div>
                0
                 
                <span>
                  downloads
                </span>
              </div>
            </div>,
            <div>
              Does stuff
            </div>,
            null,
            null,
            null,
            null,
            null,
            null,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div>
                  <span>
                    Railties
                  </span>
                  <span>
                      (v. 
                    1.0.1
                    )
                  </span>
                </div>,
                <div>
                  0
                   
                  <span>
                    downloads
                  </span>
                </div>,
              ],
              "className": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span>
                      Railties
                    </span>,
                    <span>
                        (v. 
                      1.0.1
                      )
                    </span>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Railties",
                      "className": undefined,
                    },
                    "ref": null,
                    "rendered": "Railties",
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "  (v. ",
                        "1.0.1",
                        ")",
                      ],
                      "className": undefined,
                    },
                    "ref": null,
                    "rendered": Array [
                      "  (v. ",
                      "1.0.1",
                      ")",
                    ],
                    "type": "span",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "0",
                    " ",
                    <span>
                      downloads
                    </span>,
                  ],
                  "className": undefined,
                },
                "ref": null,
                "rendered": Array [
                  "0",
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "downloads",
                      "className": undefined,
                    },
                    "ref": null,
                    "rendered": "downloads",
                    "type": "span",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Does stuff",
              "className": undefined,
            },
            "ref": null,
            "rendered": "Does stuff",
            "type": "div",
          },
          null,
          null,
          null,
          null,
          null,
          null,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className=", saveIconContainer"
            onClick={[Function]}
          >
            <FaRegGem
              className=" "
            />
          </div>,
          <div>
            <div>
              <div>
                <span>
                  Railties
                </span>
                <span>
                    (v. 
                  1.0.1
                  )
                </span>
              </div>
              <div>
                0
                 
                <span>
                  downloads
                </span>
              </div>
            </div>
            <div>
              Does stuff
            </div>
          </div>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FaRegGem
              className=" "
            />,
            "className": ", saveIconContainer",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": " ",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <div>
                  <span>
                    Railties
                  </span>
                  <span>
                      (v. 
                    1.0.1
                    )
                  </span>
                </div>
                <div>
                  0
                   
                  <span>
                    downloads
                  </span>
                </div>
              </div>,
              <div>
                Does stuff
              </div>,
              null,
              null,
              null,
              null,
              null,
              null,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div>
                    <span>
                      Railties
                    </span>
                    <span>
                        (v. 
                      1.0.1
                      )
                    </span>
                  </div>,
                  <div>
                    0
                     
                    <span>
                      downloads
                    </span>
                  </div>,
                ],
                "className": undefined,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <span>
                        Railties
                      </span>,
                      <span>
                          (v. 
                        1.0.1
                        )
                      </span>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Railties",
                        "className": undefined,
                      },
                      "ref": null,
                      "rendered": "Railties",
                      "type": "span",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          "  (v. ",
                          "1.0.1",
                          ")",
                        ],
                        "className": undefined,
                      },
                      "ref": null,
                      "rendered": Array [
                        "  (v. ",
                        "1.0.1",
                        ")",
                      ],
                      "type": "span",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "0",
                      " ",
                      <span>
                        downloads
                      </span>,
                    ],
                    "className": undefined,
                  },
                  "ref": null,
                  "rendered": Array [
                    "0",
                    " ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "downloads",
                        "className": undefined,
                      },
                      "ref": null,
                      "rendered": "downloads",
                      "type": "span",
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Does stuff",
                "className": undefined,
              },
              "ref": null,
              "rendered": "Does stuff",
              "type": "div",
            },
            null,
            null,
            null,
            null,
            null,
            null,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
